src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -323,18 +332,21 @@ public function walkJoinSource(QOM\JoinInterface $source)
         $sql .= "AND " . $this->walkJoinCondition($source->getLeft(), $source->getRight(), $source->getJoinCondition()) . " ";
         $sql .= ") "; // close on-clause
 
-        $sql .= "WHERE $leftAlias.workspace_name = ? AND $leftAlias.type IN ('" . $source->getLeft()->getNodeTypeName() ."'";
-        $subTypes = $this->nodeTypeManager->getSubtypes($source->getLeft()->getNodeTypeName());
-        foreach ($subTypes as $subType) {
-            /* @var $subType \PHPCR\NodeType\NodeTypeInterface */
-            $sql .= ", '" . $subType->getName() . "'";
+
+        if ($root) {
```

Comment:
Haven't gone much into detail about this part to be honest, but obviously made sure the functionality stays the same. Would it be adequate to leave a TODO?